### ISSUE TITLE
Can environment variables be passed in command line?

### ISSUE THREAD

**Issue by keithmork** [postmanlabs/newman#1130]
Our build system use different ports every time when deploying services, so I can't hard code the ports in environment files.

Is there any option to define/overwrite env vars in command line?

**Comment by kunagpal** [comment]
@keithmork Closing as a duplicate of #1013

**Comment by kunagpal** [comment]
@keithmork  This has been released with Newman `v3.8.0`, please see https://github.com/postmanlabs/newman#newman-run-collection-file-source-options for details on how to use this feature 😄

**Comment by keithmork** [comment]
@kunagpal
Now I use nodejs to change the environment obj and remove base path before sending request, problem solved.

The new feature is awesome, but sadly doesn't meet our needs. The hosts are defined as env vars, IIRC env vars always overwrite globals with the same name?

Thanks anyway. Great job guys : )

```
const getEnvironment = (envFilePath, extraEnvVars = []) => {
  if (!envFilePath) return null;

  const envObj = fsUtil.parseJsonFile(envFilePath);
  if (!extraEnvVars || !extraEnvVars.length) return envObj;

  extraEnvVars.forEach((envVar) => {
    const delimiterIndex = envVar.indexOf(':');
    if (delimiterIndex) {
      const key = envVar.substr(0, delimiterIndex);
      const value = envVar.substr(delimiterIndex + 1);

      for (let i = 0; i < envObj.values.length; i += 1) {
        if (envObj.values[i].key === key) {
          envObj.values[i].value = value;
          break;
        }

        if (i === envObj.values.length - 1) {
          envObj.values.push({key, value, enabled: true, type: 'text'});
        }
      }
    }
  });

  return envObj;
};
```

```
const startNewman = (commandLineArgs, callback = onDone) => {
  const options = getOptions(commandLineArgs);

  newman.run(options)
  .on('prerequest', (err, obj) => {
    if (commandLineArgs.removeBasePath) {
      obj.item.request.url.path.shift();
      // eg. API path 'health/user/register' is stored as ['health', 'user', 'register']
      // this will remove the 1st element
    }
  })
// ...
```
